ggshield/verticals/hmsl/secret_manager/hashicorp_vault.py

```diff
@@ -0,0 +1,36 @@
+import json
+import logging
+import subprocess
+
+
+logger = logging.getLogger(__name__)
+
+
+class VaultCliTokenFetchingError(Exception):
+    """Raised when the token used by vault CLI cannot be fetched."""
+
+    pass
+
+
+def get_vault_cli_token() -> str:
+    """Get the vault token used by vault CLI."""
+
+    try:
+        result = subprocess.run(
+            "vault token lookup --format=json",
+            capture_output=True,
+            text=True,
+            shell=True,
+            check=True,
+        )
+        json_content = json.loads(result.stdout)
+
+        return json_content["data"]["id"]
+    except subprocess.CalledProcessError as exc:
+        msg = "error when calling vault CLI."
```

Comment:
Nit: should Vault be written with a capital V ? (For all messages)

Reply:
We could indeed put Vault everywhere, it seems to be the naming convention used on the Hashicorp documentation even when referring to the CLI binary (like [here](https://developer.hashicorp.com/vault/docs/commands)).